### ISSUE TITLE
Parameter typing missing requestBody in actions.ts

### ISSUE THREAD

**Issue by leewardbound** [piercefreeman/mountaineer#94]
Hi there -

The generated parameter typing is incorrect (invalid) on my action in actions.ts - it should include a `requestBody` that matches my `data` parameter type (AdminTableItem); the variable gets used in the body of the function, but is not present in the parameters (red underline on right half of screenshot). If I manually edit the actions.ts to add the proper parameter, my action functions as expected.

I believe the problem arrises when using a controller URL with `/{item_id}` arguments in it.

![image](https://github.com/piercefreeman/mountaineer/assets/395107/d8e09dc0-8baf-4efb-b1ef-3dc6d9ea10b2)

I tried tracing this down and got some debug logging in `build_actions.py`, only to find that this `schema_ref` is null for my action for some reason. I'm sure I'm using something wrong to cause `schema_ref` to be null, but since it causes broken TS to generate, maybe mountaineer should throw an exception if I try to register an action with a null schema ref?

https://github.com/piercefreeman/mountaineer/blob/main/mountaineer/client_builder/build_actions.py#L147

**Comment by piercefreeman** [comment]
@leewardbound Your action definition looks fine to me. The `data` param should resolve as the requestBody and the other two args should be URL params. What is the definition of your `AdminTableItem`? And what version of Mountaineer are you using?

**Comment by leewardbound** [comment]
Aha, this makes sense - Here's 3 types, I was using the first 2, and the 3rd one fixed it -

```
AnyType = None | bool | str | int | datetime | UUID
AdminTableItem = dict[str, AnyType]

class AdminTableRow(BaseModel):
    data: AdminTableItem
```

+ My AdminTableItem type isn't a Pydantic model (this was my b, I confused it for the AdminTable type which is Pydantic)
+ I've been able to fix it by wrapping the AdminTableItem in a Pydantic model
+ Using AdminTableRow (from the following definition) produces valid actions.ts


Thanks again Pierce! Closing as a non-issue.

**Comment by piercefreeman** [comment]
@leewardbound In the second commit of this PR, I've added a more verbose exception for similar cases. However in tracing this code path it seems like this case was always supposed to raise. Did you not see that Exception when you were initially testing with the `AdminTableItem` typedef?

https://github.com/piercefreeman/mountaineer/pull/95

**Comment by leewardbound** [comment]
I did not see that exception! Though I have noticed sometimes I miss errors in my console due to the reloader's verbosity, so let's chalk it up to user error 😊

**Comment by piercefreeman** [comment]
Great. If you see it again, do raise another flag. I have a logger refactor in the works so hopefully that helps some of the (unnecessary) verbosity in the happy path.

**Comment by piercefreeman** [comment]
FYI @leewardbound, the logging refactor was merged in https://github.com/piercefreeman/mountaineer/pull/102 and available from [v0.4.2.dev3](https://github.com/piercefreeman/mountaineer/releases/tag/v0.4.2.dev3) onward. It should make build errors more verbose and dial down the repetition of other logging messages.
